codec-redis/src/main/java/io/netty/handler/codec/redis/RedisDecoder.java

```diff
@@ -0,0 +1,299 @@
+     * @param maxBulkStringChunkSize the maximum length of bulk string chunk length.
+     * @param maxInlineMessageLength the maximum length of inline message.
+     */
+    public RedisDecoder(int maxBulkStringChunkSize, int maxInlineMessageLength) {
+        if (maxBulkStringChunkSize <= 0 || maxBulkStringChunkSize > REDIS_MESSAGE_MAX_LENGTH) {
+            throw new RedisCodecException("maxBulkStringChunkSize: " + maxBulkStringChunkSize +
+                                          " (expected: <= " + REDIS_MESSAGE_MAX_LENGTH + ")");
+        }
+        if (maxInlineMessageLength <= 0 || maxInlineMessageLength > REDIS_MESSAGE_MAX_LENGTH) {
+            throw new RedisCodecException("maxInlineMessageLength: " + maxInlineMessageLength +
+                                          " (expected: <= " + REDIS_MESSAGE_MAX_LENGTH + ")");
+        }
+        this.maxInlineMessageLength = maxInlineMessageLength;
+        this.maxBulkStringChunkSize = maxBulkStringChunkSize;
+    }
+
+    @Override
+    protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) throws Exception {
+        try {
+            for (;;) {
+                switch (state) {
+                case DECODE_TYPE:
+                    if (!decodeType(in)) {
+                        return;
+                    }
+                    break;
+                case DECODE_INLINE:
+                    if (!decodeInline(in, out)) {
+                        return;
+                    }
+                    break;
+                case DECODE_LENGTH:
+                    if (!decodeLength(in, out)) {
+                        return;
+                    }
+                    break;
+                case DECODE_BULK_STRING:
+                    if (!decodeBulkString(in, out)) {
+                        return;
+                    }
+                    break;
+                default:
+                    throw new RedisCodecException("Unknown state: " + state);
+                }
+            }
+        } catch (RedisCodecException e) {
+            resetDecoder();
+            throw e;
+        } catch (Exception e) {
+            resetDecoder();
+            throw new RedisCodecException(e);
+        }
+    }
+
+    private void resetDecoder() {
+        state = State.DECODE_TYPE;
+        expectedBulkLength = 0;
+        toReadBulkLength = 0;
+    }
+
+    private boolean decodeType(ByteBuf in) throws Exception {
+        if (!in.isReadable()) {
+            return false;
+        }
+        type = RedisMessageType.valueOf(in.readByte());
+        state = type.isInline() ? State.DECODE_INLINE : State.DECODE_LENGTH;
+        return true;
+    }
+
+    private boolean decodeInline(ByteBuf in, List<Object> out) throws Exception {
+        ByteBuf lineBytes = readLine(in);
+        if (lineBytes == null) {
+            if (in.readableBytes() > maxInlineMessageLength) {
+                throw new RedisCodecException("length: " + in.readableBytes() +
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "length: " 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1192)